### PR TITLE
Become independent from the user component

### PR DESCRIPTION
**Pull request by rjzondervan**
- Refactor of user controller to login without user component
- Refactor of user token authenticator to verify tokens without the user component

### CODE REVIEW

File: api/composer.lock [hunk 1]
```diff
@@ -14782,6 +14782,7 @@
     "minimum-stability": "stable",
     "stability-flags": {
         "common-gateway/formio-bundle": 20,
+        "common-gateway/naamgebruik-vrijbrp-bundle": 20,
```
Comment (WilcoLouwerse) [inline api/composer.lock:14785]: We don't want this into development branch right?

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

File: api/src/Entity/Application.php [hunk 2]
```diff
@@ -488,4 +496,22 @@ public function prePersist()
...
+     */
+    public function getPrivateKey(): ?string
+    {
+        return $this->privateKey;
+    }
+
+    /**
+     * @param string|null $privateKey
+     */
+    public function setPrivateKey(?string $privateKey): self
```
[the remark this answers is not in the document]
Comment (rjzondervan) [reply api/src/Entity/Application.php:511]: These are auto generated docblocks, usually we don't have docblocks in entities

File: api/src/Entity/Application.php [hunk 3]
```diff
@@ -488,4 +496,22 @@ public function prePersist()
             $this->setPublic($secret);
         }
     }
+
+    /**
+     * @return string|null
+     */
+    public function getPrivateKey(): ?string
```
[the remark this answers is not in the document]
Comment (rjzondervan) [reply api/src/Entity/Application.php:503]: These are auto generated docblocks, usually we don't have docblocks in entities

File: api/src/Security/TokenAuthenticator.php [hunk 4]
```diff
@@ -108,84 +102,6 @@ public function validateToken(string $token): array
...
-            $parentOrganizations = $this->getParentOrganizations($parentOrganizations, $organization, $this->commonGroundService, $this->functionService);
-        }
-        $organizations[] = 'localhostOrganization';
-        $parentOrganizations[] = 'localhostOrganization';
-        $this->session->set('organizations', $organizations);
-        $this->session->set('parentOrganizations', $parentOrganizations);
-        $this->session->set('ActiveOrganization', $user['organization'] ?? $this->applicationService->getApplication()->getOrganization());
-    }
-
     private function prefixRoles(array $roles): array
```
[the remark this answers is not in the document]
Comment (rjzondervan) [reply api/src/Security/TokenAuthenticator.php:105]: This is old code that didn't have a docblock yet
Will fix in a later PR

File: api/src/Controller/UserController.php [hunk 5]
```diff
@@ -47,51 +50,38 @@ public function LoginAction(Request $request, CommonGroundService $commonGroundS
     /**
      * @Route("api/users/login", methods={"POST"})
      */
-    public function apiLoginAction(Request $request, CommonGroundService $commonGroundService, FunctionService $functionService)
+    public function apiLoginAction(Request $request, UserPasswordHasherInterface $hasher, SerializerInterface $serializer, \CommonGateway\CoreBundle\Service\AuthenticationService $authenticationService)
```
[the remark this answers is not in the document]
Comment (rjzondervan) [reply api/src/Controller/UserController.php:53]: There are two authenticationservices at the moment, both used in this code. So yes, would like to, but that would need another round of checking.

File: api/src/Controller/UserController.php [hunk 6]
```diff
@@ -47,51 +50,38 @@ public function LoginAction(Request $request, CommonGroundService $commonGroundS
...
+            return new Response(json_encode($response), 401, ['Content-type' => 'application/json']);
         }
 
-        // Set orgs in session for multitenancy
-        // Get user object with userGroups (login only returns a user with userGroups as: /groups/uuid)
-        $user = $commonGroundService->getResource(['component' => 'uc', 'type' => 'users', 'id' => $userLogin['id']], [], false);
-        $organizations = [];
-        if (isset($user['organization'])) {
-            $organizations[] = $user['organization'];
+        if ($user->getOrganisation() !== null) {
```
[the remark this answers is not in the document]
Comment (rjzondervan) [reply api/src/Controller/UserController.php:70]: In this case it does not matter what is returned, only if something has been returned.

File: api/src/Controller/UserController.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (WilcoLouwerse) [reply api/src/Controller/UserController.php:53]: You could do a use \CommonGateway\CoreBundle\Service\AuthenticationService as SomeNameYouLike

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**